File: 263-ugly-number/ugly-number.py
class Solution:
    def isUgly(self, n: int) -> bool:
        
        arr = [2,3,5]
        def helper(n):
            if n ==1:
                return True
            if n == 0:
                return False
            for f in arr:
                div = n//f
                if n%f ==0:
                    return True  and helper(div)
            return False

        return helper(n)

File: 263-ugly-number/ugly-number.c
int arr[3] = {2,3,5};
bool helper(int n) {
    if (n == 1)
        return true;
    if (n == 0) 
        return false;
    for( int i = 0; i < 3; ++i) {
        int div = n / arr[i];
        if(n%arr[i] == 0)
        return true && helper(div);
    }
    return false;
};

bool isUgly(int n) {
    return helper(n);
}